mne/channels/channels.py

```diff
@@ -206,6 +208,90 @@ def equalize_channels(instances, copy=True, verbose=None):
     return equalized_instances
 
 
+def unify_bad_channels(insts):
+    """Unify bad channels across a list of instances, using the union.
+
+    Parameters
+    ----------
+    insts : list
+        List of instances (:class:`~mne.io.Raw`, :class:`~mne.Epochs`,
+        :class:`~mne.Evoked`, :class:`~mne.time_frequency.Spectrum`,
+        :class:`~mne.time_frequency.EpochSpectrum`) across which to unify bad channels.
+
+    Returns
+    -------
+    insts : list
+        List of instances with bad channels unified across instances.
+
+    See Also
+    --------
+    mne.channels.equalize_channels
+    mne.channels.rename_channels
+    mne.channels.combine_channels
+
+    Notes
+    -----
+    This function modifies the instances in-place.
+
+    .. versionadded:: 1.6
+    """
+    from ..utils import _validate_type
+
+    from ..io import BaseRaw
+    from ..epochs import Epochs
+    from ..evoked import Evoked
+    from ..time_frequency.spectrum import BaseSpectrum
+
+    # check that input is a list
+    _validate_type(insts, (list, tuple), "instance type")
+
+    # check input is not an empty list
+    if len(insts) == 0:
+        raise ValueError("Be sure insts is not empty list")
+
+    # first check that each object is mne object
+    inst_type = type(insts[0])
+    valid_types = (BaseRaw, Epochs, Evoked, BaseSpectrum)
+    for inst in insts:
+        _validate_type(inst, valid_types, "instance type")
+        if type(inst) != inst_type:
+            raise ValueError("All insts must be the same type")
+
+    # check that all channels have the same name and same number
+    ch_names = insts[0].ch_names
+    diff_chns = []
+    for inst in insts[1:]:
+        if inst.ch_names != ch_names:
+            dif = set(inst.ch_names) ^ (set(ch_names))
+            diff_chns.extend(list(dif))
+            if ch_names.sort() == inst.ch_names.sort():
+                raise ValueError(
+                    "Channel names are sorted differently across"
+                    "instances. Please use"
+                    "mne.channels.equalize_channels."
+                )
+
+    if len(diff_chns) > 0:
+        raise ValueError(
+            "Channel names are not consistent across instances. Be sure"
+            "consistent naming is used across channels. Mismatched channels"
+            f"are {diff_chns}"
+        )
```

Comment:
```suggestion
    for inst in insts[1:]:
        dif = set(inst.ch_names) ^ set(ch_names)
        if len(dif):
            # TODO raise error, we know some chs need to be dropped
            raise ValueError("")
        elif inst.ch_names != ch_names:
            raise ValueError(
                "Channel names are sorted differently across instances. Please use "
                "mne.channels.equalize_channels."
            )
```
this refactor does 2 things:

1. fail early / don't bother checking all insts if we know at least one is mismatched
2. reduces cyclomatic complexity (i.e., number of intents from for/if/while clauses), which makes the code easier to read  and reason about